core/controllers/base.py

```diff
@@ -140,24 +140,25 @@ def __init__(self, request, response):  # pylint: disable=super-init-not-called
         # Initializes the return dict for the handlers.
         self.values = {}
 
-        self.user_id = current_user_services.get_current_user_id()
+        self.gae_user_id = current_user_services.get_current_gae_user_id()
+        self.user_id = None
         self.username = None
         self.partially_logged_in = False
 
-        if self.user_id:
-            user_settings = user_services.get_user_settings(
-                self.user_id, strict=False)
+        if self.gae_user_id:
+            user_settings = user_services.get_user_settings_by_gae_user_id(
```

Comment:
I suggest keeping the "strict" semantics in the new method, in the spirit of doing the minimal change.

Reply:
Done.

Reply:
Done.